op-acceptance-tests/tests/sync_tester/hardforks_ext/hardforks_ext.go

```diff
@@ -0,0 +1,199 @@
+	L1CLBeaconEndpoint = getEnvOrDefault("L1_CL_BEACON_ENDPOINT", DefaultL1CLBeaconEndpoint)
+	L1ELEndpoint       = getEnvOrDefault("L1_EL_ENDPOINT", DefaultL1ELEndpoint)
+)
+
+// getEnvOrDefault returns the environment variable value or the default if not set
+func getEnvOrDefault(envVar, defaultValue string) string {
+	if value := os.Getenv(envVar); value != "" {
+		return value
+	}
+	return defaultValue
+}
+
+// getEnvUint64OrDefault returns the environment variable value as uint64 or the default if not set
+func getEnvUint64OrDefault(envVar string, defaultValue uint64) uint64 {
+	if value := os.Getenv(envVar); value != "" {
+		if parsed, err := strconv.ParseUint(value, 10, 64); err == nil {
+			return parsed
+		}
+	}
+	return defaultValue
+}
+
+// setupOrchestrator initializes and configures the orchestrator for the test
+func setupOrchestrator(gt *testing.T, t devtest.T, blk uint64) *sysgo.Orchestrator {
+	l := t.Logger()
+
+	// Override configuration with Tailscale endpoints if Tailscale networking is enabled
+	l2ELEndpoint := L2ELEndpoint
+	l1CLBeaconEndpoint := L1CLBeaconEndpoint
+	l1ELEndpoint := L1ELEndpoint
+
+	if os.Getenv("TAILSCALE_NETWORKING") == "true" {
+		l2ELEndpoint = getEnvOrDefault("L2_EL_ENDPOINT_TAILSCALE", DefaultL2ELEndpointTailscale)
+		l1CLBeaconEndpoint = getEnvOrDefault("L1_CL_BEACON_ENDPOINT_TAILSCALE", DefaultL1CLBeaconEndpointTailscale)
+		l1ELEndpoint = getEnvOrDefault("L1_EL_ENDPOINT_TAILSCALE", DefaultL1ELEndpointTailscale)
+	}
+
+	// Setup orchestrator directly without TestMain
+	logger := testlog.Logger(gt, log.LevelInfo)
+	onFail := func(now bool) {
+		if now {
+			gt.FailNow()
+		} else {
+			gt.Fail()
+		}
+	}
+	onSkipNow := func() {
+		gt.SkipNow()
+	}
+	p := devtest.NewP(context.Background(), logger, onFail, onSkipNow)
+	gt.Cleanup(p.Close)
+
+	// Runtime configuration values
+	l.Info("Runtime configuration values for TestSyncTesterExtEL")
+	l.Info("L2_NETWORK_NAME", "value", L2NetworkName)
+	l.Info("L1_CHAIN_ID", "value", L1ChainID)
+	l.Info("L2_EL_ENDPOINT", "value", l2ELEndpoint)
+	l.Info("L1_CL_BEACON_ENDPOINT", "value", l1CLBeaconEndpoint)
+	l.Info("L1_EL_ENDPOINT", "value", l1ELEndpoint)
+	l.Info("TAILSCALE_NETWORKING", "value", os.Getenv("TAILSCALE_NETWORKING"))
+
+	// Create orchestrator with the same configuration that was in TestMain
+	opt := sysgo.DefaultMinimalExternalELSystemWithEndpointAndSuperchainRegistry(&sysgo.DefaultMinimalExternalELSystemIDs{}, l1CLBeaconEndpoint, l1ELEndpoint, l2ELEndpoint, L1ChainID, L2NetworkName, eth.FCUState{
+		Latest:    blk,
+		Safe:      blk,
+		Finalized: blk,
+	})
+
+	orch := sysgo.NewOrchestrator(p, stack.SystemHook(opt))
+	stack.ApplyOptionLifecycle[*sysgo.Orchestrator](opt, orch)
+
+	return orch
+}
+
+func SyncTesterHFSExt(gt *testing.T, upgradeName string, forkTimestamp func(net *dsl.L2Network) *uint64) {
+	t := devtest.SerialT(gt)
+	l := t.Logger()
+
+	// Initial block number to sync from before the upgrade
+	blk := networkUpgradeBlocks[upgradeName] - 5
```

Comment:
Would be nice to check the hardcode upgraded bock number is correct, by fetching the block using the blocknumber and comparing with the timestamp from rollup config.